### PR TITLE
feat: Unblocked C2 Upload submission on fees register api error (FPLA-1328)

### PR DESCRIPTION
**Pull request by pd-dudkiewicz**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-1328

### Change description ###

- Unblocked C2 Upload submission on fees register api error
- Added making payments only when fee was displayed

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionController.java [hunk 1]
```diff
@@ -164,7 +162,7 @@ public void handleSubmittedEvent(
         @RequestBody @NotNull CallbackRequest callbackRequest) {
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
-        if (featureToggleService.isPaymentsEnabled()) {
+        if (featureToggleService.isPaymentsEnabled() && displayAmountToPay(caseDetails)) {
```
Comment (akomisarek) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionController.java]: Thanks for taking care of this onoe as well.

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionController.java]: This should resolve the purple banner stuff right?

File: ccd-definition/CaseEventToFields/uploadC2/uploadC2-PREPARE_FOR_HEARING-nonprod.json [hunk 2]
```diff
@@ -15,12 +15,25 @@
...
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseEventID": "uploadC2-PREPARE_FOR_HEARING",
+    "CaseFieldID": "amountToPay",
+    "PageFieldDisplayOrder": 3,
+    "DisplayContext": "READONLY",
+    "PageID": 2,
+    "PageDisplayOrder": 2,
+    "PageColumnNumber": 1,
+    "FieldShowCondition": "displayAmountToPay=\"Yes\"",
     "CallBackURLMidEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/upload-c2/validate-pba-number/mid-event"
```
[the remark this answers is not in the document]
Comment (pd-dudkiewicz) [reply ccd-definition/CaseEventToFields/uploadC2/uploadC2-PREPARE_FOR_HEARING-nonprod.json]: yup, good spot, moved :+1:

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java [hunk 3]
```diff
@@ -56,27 +57,25 @@
...
         //workaround for previous-continue bug
         if (shouldRemoveDocument(caseData)) {
             removeDocumentFromData(data);
         }
 
-        List<String> errors = new ArrayList<>();
-        if (featureToggleService.isFeesEnabled()) {
-            try {
+        try {
+            if (featureToggleService.isFeesEnabled()) {
```
[the remark this answers is not in the document]
Comment (pd-dudkiewicz) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java]: I was thinking about that, we could set it to "no" before try-catch block and then set it to "yes" only when needed. But I think it's better to leave it as it, so it's easy to see how we're handling FeeRegisterException

**Pull request merged**